projects/Mexico/mexico_model.py

```diff
@@ -0,0 +1,131 @@
+import argparse
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+
+from tabgpt.callbacks import whole_epoch_train_loss
+from tabgpt.data.mexico.data_setup import MexicoData
+from tabgpt.data.mexico.data_setup_weekly import MexicoDataWeekly
+from tabgpt.utils import evaluation, predict
+import torch
+from torch.utils.data import TensorDataset, DataLoader
+
+from tabgpt.model_hf import tabGPT_HF, tabGPTConfig
+from tabgpt.model import tabGPT
+from tabgpt.trainer import Trainer
+from tabgpt.col_embed import Embedder
+
+from IPython import embed
+
+
+if torch.cuda.is_available():       
+    device = torch.device("cuda:0")
+    print("Using GPU.")
+else:
+    print("No GPU available, using the CPU instead.")
+    device = torch.device("cpu")
+
+
+def plot_timeseries(df, suffix, include_preds=False):
+    if include_preds:
+        ts = df.groupby(['date'])[['quantity', 'yhat']].sum().reset_index()
+    else:
+        ts = df.groupby(['date'])['quantity'].sum().reset_index()
+    plt.figure()
+    ts.index = ts['date']
+    ts['quantity'].plot(style='r', label="quantity")
+    if include_preds:
+        ts['yhat'].plot(style='b-.', label="predictions")
+    plt.legend(fontsize=15)
+    plt.ylabel("sum", fontsize=15)
+    plt.tight_layout()
+    plt.savefig("ts_{}.png".format(suffix))
+    plt.clf()
+
+
+def main(test, pretrained):
+    np.random.seed(666)
+    torch.manual_seed(42)
+
+    mexico = MexicoData()
+    # mexico = MexicoDataWeekly()
+    mexico.setup()
+    n_cols = mexico.n_features
+
+    embedder = Embedder(mexico)
+
+    embedder.train()
+    features_embeds_train = embedder.embed(n_cols)
+
+    embedder.val()
+    features_embeds_val = embedder.embed(n_cols)
+
+    # embedder.test()
+    # features_embeds_test = embedder.embed(n_cols)
+
+    df_train = mexico.df_train
+    df_val = mexico.df_val
+    # df_test = mexico.df_test
+
+
+    train_dataset = TensorDataset(
+        features_embeds_train,
+        torch.tensor(df_train[mexico.target_column].tolist(), dtype=torch.float32)
```

Comment:
use `mexico.main_target`

Reply:
Perhaps also changing all other usages of `mexico.target_column` too?